src/formatDate.js

```diff
@@ -50,7 +50,39 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let arrDate = [];
+  const copyFromFormat = Array.from(fromFormat);
+  let result = '';
+
+  arrDate = date.split(fromFormat[3]);
+
+  if (copyFromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    const index = copyFromFormat.indexOf('YYYY');
+
+    copyFromFormat[index] = 'YY';
+    arrDate[index] = arrDate[index].slice(2);
+  } else {
+    const index = copyFromFormat.indexOf('YY');
+
+    copyFromFormat[index] = 'YYYY';
+
+    if (arrDate[index] < 30) {
+      arrDate[index] = 20 + arrDate[index];
+    } else {
+      arrDate[index] = 19 + arrDate[index];
+    }
+  }
```

Comment:
Instead of doing all these actions you can create object with keys from "from format" and values from your "arrDate". Then you can make only 3 test for your "obj.YYYY" or "obj.YY", which will start only if fromFormat and toFormat year records are different